### DISCUSSION TITLE
conceptually get multiple server instances working together

### DISCUSSION THREAD

**Discussion by ckalas** [miguelgrinberg/python-socketio#1025] [Q&A]
I am trying to integrate this into a FastAPI application as an ASGI sub-application. It is deployed on Kubernetes and everything works fine with a single replica; as soon as i increase that it fails. I had thought by passing the redis server running on the cluster to the `client_manager` kwarg to the `socketio.AsyncServer` this would work, but i always get errors like below.

```  
mgr = socketio.AsyncRedisManager(
        f"redis://{os.getenv('REDIS_HOST', 'localhost')}"
    )
sio = socketio.AsyncServer(
        cors_allowed_origins="*", async_mode="asgi", client_manager=mgr
    )
# Invalid session hH5knPe_GeTb_i3fAAAA (further occurrences of this error will be logged with level INFO)```

Could this conceptually work or am i missing something?

**Comment by miguelgrinberg** [comment]
The issues board is for issues. Please do not post questions about your application there, that's what the discussions board is for.

Your load balancer needs to use sticky sessions for this to work. See the documentation for details.
